queue/abstract.lua

```diff
@@ -570,15 +570,43 @@ function method.start()
     end
 
     for _, tube_tuple in _queue:pairs() do
-        local tube = recreate_tube(tube_tuple)
-        -- gh-66: release all taken tasks on start
-        tube_release_all_tasks(tube)
+        -- Recreate tubes for registered drivers only.
+        if queue.driver[tube_tuple[4]] ~= nil then
+            local tube = recreate_tube(tube_tuple)
+            -- gh-66: release all taken tasks on start
+            tube_release_all_tasks(tube)
+        end
     end
 
     session.on_disconnect(queue._on_consumer_disconnect)
     return queue
 end
 
+--- Register the custom driver.
+-- Unlike the "register_driver" method from init.lua, this method
+-- recreates the existing tubes of the registered driver.
+function method.register_driver(driver_name, tube_ctr)
```

Comment:
I would change function name to something like 'recreate_tubes()' to avoid confusion with `register_driver()` in `init.lua`.

Reply:
Sorry, but it's impossible. For the client, this is the same method as before calling box.cfg ().